### DISCUSSION TITLE
Paper Cross et al 2017

### DISCUSSION THREAD

**Discussion by reepaes** [mtex-toolbox/mtex#2034] [Ask Anything] [answered]
I am trying to follow the concept proposed by Cross et al. 2017, to separate the relict grains and recrystallized grains, but I am facing some problems with the function "tradeoff", in the following step:

% Use trade-off curve to find the cutoff between well-constrained and  poorly-constrained grains
knee = tradeOff(fraction);
    xlabel('Number of grains (cumulative)')
    ylabel('Indexed fraction')

the answer showed is: 
Unrecognized function or variable 'tradeOff'

I  tried to change for different forms like "TradeOff" "Tradeoff" "tradeoff" "tradeOff", and looked on Matlab support about this function and I didn't find anything also, if someone could help me I would really appreciate it
Thanks

**Comment by zmichels** [comment]
There is not a function named tradeoff in MTEX or MATLAB. Could the function in question be code that someone else wrote and passed along to you?

**Reply by reepaes** [reply]
Thank you for the quick answer.
Is a script related to this paper : https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2017GL073836
Now that I know that is not related to MTEX or Matlab I will ask to the authors about it.
All the best

**Reply by zmichels** [reply, marked as answer]
Sounds good. I am familiar with the paper and have helped folks in the past to implement the approach in MTEX with custom code. Let us know if we can help further.
The error you are getting seems to suggest that the function is not present on the/your system path where it's being called. If you've used the function previously, perhaps the system has changed since then (update to MTEX or MATLAB path?), or perhaps it is a separate/new MATLAB installation... and may require putting a copy of the tradeoff.m function in your Matlab path. 

Best wishes,
Zach

**Comment by AzdiarGazder** [comment]
Hi @reepaes,

Please have a look at the following [weblink](https://github.com/AzdiarGazder/mtexTools/tree/main/rcrxSegmenter) that could be of use to you. 
It implements a solution that enables the segmenting of recrystallised grains.

Hope it helps.

Warm regards,
Azdi

**Reply by reepaes** [reply]
Thank you Azdi
I will check it. 
Best
Renata
